web_widget_x2many_2d_matrix/static/src/js/widget_x2many_2d_matrix.js

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2015 Holger Brunn <hbrunn@therp.nl>
+ * Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+ * Copyright 2018 Simone Orsi <simone.orsi@camptocamp.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('web_widget_x2many_2d_matrix.widget', function (require) {
+  "use strict";
+
+  var core = require('web.core');
+  // var FieldManagerMixin = require('web.FieldManagerMixin');
+  var field_registry = require('web.field_registry');
+  var relational_fields = require('web.relational_fields');
+  var weContext = require('web_editor.context');
+  // var Helpers = require('web_widget_x2many_2d_matrix.helpers');
+  var AbstractField = require('web.AbstractField');
+  var X2Many2dMatrixRenderer = require('web_widget_x2many_2d_matrix.X2Many2dMatrixRenderer');
+
+  var WidgetX2Many2dMatrix = relational_fields.FieldOne2Many.extend({
+    widget_class: 'o_form_field_x2many_2d_matrix',
+    /**
+     * Initialize the widget & parameters.
+     *
+     * @param {Object} parent: contains the form view.
+     * @param {String} name: the name of the field.
+     * @param {Object} record: Contains the information about the database records.
+     * @param {Object} options: Contains the view options.
+     */
+    init: function (parent, name, record, options) {
+        this._super(parent, name, record, options);
+        this.init_params();
+    },
+
+    /**
+     * Initialize the widget specific parameters.
+     * Sets the axis and the values.
+     */
+    init_params: function () {
+        var node = this.attrs;
+        this.by_x_axis = {};
+        this.by_y_axis = {};
+        this.field_x_axis = node.field_x_axis || this.field_x_axis;
+        this.field_y_axis = node.field_y_axis || this.field_y_axis;
+        this.field_label_x_axis = node.field_label_x_axis || this.field_x_axis;
+        this.field_label_y_axis = node.field_label_y_axis || this.field_y_axis;
+        this.x_axis_clickable = this.parse_boolean(node.x_axis_clickable || '1');
+        this.y_axis_clickable = this.parse_boolean(node.y_axis_clickable || '1');
+        this.field_value = node.field_value || this.field_value;
+        // TODO: is this really needed? Holger?
```

Comment:
yes, I'd have asked about this at the end of the review: This seems to be a bit orthogonal to using the renderer, but this property allows you to pull some nice tricks like making specific fields in the matrix readonly and the like. This worked because before, you could set attributes directly on the html input element. I doubt there's an elegant way to do this with the matrix renderer (maybe something with simulating an attrs dict?), if so, please update the readme and move this to the roadmap, maybe someone has a good idea how to implement this. I wouldn't block on this I think

Reply:
PS: I think if you can query those values and make them attributes of the table cell of the data point in question, this should be rather simple to do and we don't lose features in the migration. The good place would be https://github.com/OCA/web/pull/871/files#diff-d6a628d507f069353237db3c362415ccR218 - I think in most cases this was used to either set a class or make fields readonly/disabled. As the disabled property doesn't cascade, it wouldn't be a solution for this case, but where that's necessary an author could mark cells that are supposed to be disabled and put some small script on top of the widget to do so. Or patch an extra disabled field into the widget, but I don't think we need this in version 1

Reply:
yep, adding data attrs in there is fine. I'll add it to the roadmap and put a comment here.